core/container.go

```diff
@@ -258,6 +261,25 @@ func (container *Container) From(ctx context.Context, gw bkgw.Client, addr strin
 		imgSpec.Config.Env = newEnv
 		return imgSpec.Config
 	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	payload, err = ctr.ID.decode()
+	if err != nil {
+		return nil, err
+	}
+
+	fs := imgSpec.RootFS
+	if len(fs.DiffIDs) > 0 {
```

Comment:
I think the digest you're looking for is actually returned already as the first discarded value on this line:

```go
_, cfgBytes, err := gw.ResolveImageConfig(ctx, ref, llb.ResolveImageConfigOpt{
```

I think you could tweak it like this:

```go
	ref := reference.TagNameOnly(refName)

	digest, cfgBytes, err := gw.ResolveImageConfig(ctx, ref.String(), llb.ResolveImageConfigOpt{
		// ...
	})
	if err != nil {
		return nil, err
	}

	digested, err := reference.WithDigest(ref, digest)
	if err != nil {
		return nil, err
	}
```

and then store `digested.String()` as the `ImageRef`.

Reply:
fixed